applications/nrf5340_audio/src/audio_module/src/aobj_api.h

```diff
@@ -0,0 +1,494 @@
+ */
+int amod_data_rx(struct amod_handle *handle, struct ablk_block *block, k_timeout_t timeout);
+
+/**
+ * @brief Send a block to a module or the first in a sequence of modules. The block
+ *        is processed within the module or sequence of modules. The result is returned
+ *        via the final module's output FIFO. module(s) and returned from the final module.
+ *        All the input data is consumed within the call and thus the input data buffer
+ *        maybe released once the function retrurns.
+ *
+ * @param handle_tx   A handle to the module to send the input block to
+ * @param handle_rx   A handle to the module to receive a block from
+ * @param block_tx    Pointer to the audio data block to send
+ * @param block_rx    Pointer to the audio data block received
+ * @param timeout     Non-negative waiting period to wait for operation to complete
+ *	                  (in milliseconds). Use K_NO_WAIT to return without waiting,
+ *	                  or K_FOREVER to wait as long as necessary.
+ *
+ * @return 0 if successful, error otherwise
+ */
+int amod_data_tx_rx(struct amod_handle handle_tx, struct amod_handle handle_rx,
+		    struct ablk_block *block_tx, struct ablk_block *block_rx, k_timeout_t timeout);
+
+/**
+ * @brief Helper function to configure the module's parameter structure with
+ *        the module's description and thread information..
+ *
+ * @param parameters   Pointer to the module set-up parameters
+ * @param description  Pointer to the module's private description
+ * @param stack        Memory block for the threads stack
+ * @param stack_size   Size of the threads stack
+ * @param priority     Priority of the thread insatnce, one of #amod_id
+ * @param msg_rx       Number of concurrent input messages
+ * @param msg_tx       Number of concurrent output messages
+ * @param data_slab    Slab to take a data item from
+ * @param data_size    Size of the data item that will be taken
+ *
+ * @return 0 if successful, error otherwise
+ */
+int amod_parameters_configure(struct amod_parameters *parameters,
+			      struct amod_description *description, k_thread_stack_t *stack,
+			      size_t stack_size, int priority, struct data_fifo *msg_rx,
+			      struct data_fifo *msg_tx, struct k_mem_slab *data_slab,
+			      size_t data_size);
+
+/**
+ * @brief Helper function to get the base and instance names for a given
+ *        module handle.
+ *
+ * @param handle         A handle to the module instance
+ * @param base_name      A pointer to the name of the module
+ * @param instance_name  A pointer to the name of the current module instance
+ *
+ * @return 0 if successful, error otherwise
+ */
+int amod_names_get(struct amod_handle *handle, char *base_name, char *instance_name);
+
+/**
+ * @brief Helper function to get the state of a given module handle.
+ *
+ * @param handle  A handle to the module instance
+ *
+ * @return The state of the module if successful, error otherwise
```

Comment:
Should have a note about the return being positive for state and negative otherwise.